sw/host/opentitantool/src/command/rsa.rs

```diff
@@ -98,10 +106,126 @@ impl CommandDispatch for RsaKeyGenerateCommand {
     }
 }
 
+/// Export public information of a private or public RSA key
+/// to a C header that can be used in the ROM or ROM_EXT
+#[derive(Debug, StructOpt)]
+pub struct RsaKeyExportCommand {
+    #[structopt(
+        name = "DER_FILE",
+        help = "RSA public or private key file in DER format"
+    )]
+    der_file: PathBuf,
+    #[structopt(name = "OUTPUT_FILE", help = "output header file to generate")]
+    output_file: Option<PathBuf>,
+}
+
+/// Write the content of a big integer to a big as an array of 32-bit words.
+/// The number must be represented as an array of bytes in little-endian whose
+/// length is a multiple of four. The output is compatible with the format used
+/// by the ROM and ROM_EXT with sigverify_rom_key_t, ie the modulus and n0_inv
+/// are represented as an array of 32-bit words in little-endian. To make the function's
+/// output flexible, the function can print up to a specified number of items per
+/// line, and each line can be prefixed and suffixed with a specified string
+fn write_bigint_as_u32<W: Write>(
+    out: &mut W,
+    number: Vec<u8>,
+    nr_per_line: usize,
+    prefix: &str,
+    suffix: &str,
+) -> Result<()> {
+    let chunk = std::mem::size_of::<u32>();
+    assert!(
+        number.len() % chunk == 0,
+        "the big integer size is not a multiple of 4 bytes"
+    );
+    for (idx, num) in number.windows(chunk).step_by(chunk).enumerate() {
+        // print prefix and newline on multiples of nr_per_line
+        if idx % nr_per_line == 0 {
+            if idx != 0 {
+                write!(out, "{}\n", suffix)?;
+            }
+            write!(out, "{}", prefix)?;
+        }
+        // print one 32-bit word
+        let val =
+            num[0] as u32 | (num[1] as u32) << 8 | (num[2] as u32) << 16 | (num[3] as u32) << 24;
+        write!(out, "{:#010x}, ", val)?;
+    }
+    // extra return to the line
+    write!(out, " {}\n", suffix)?;
+    Ok(())
+}
```

Comment:
`write_bigint_as_u32` could benefit from a unit test. The test would also be helpful as an example (i.e. it would show an example input and an example output).

Reply:
Yes, I will add one to the next update of the PR.

Reply:
I have added a unittest here (I don't think it is necessary to move this function to the opentitanlib?).
Hopefuly the test clarifies what the function is doing.